Detectors/MUON/MCH/Simulation/src/Response.cxx

```diff
@@ -0,0 +1,99 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/** @file Response.cxx
+ * C++ MCH charge induction and signal generation incl. Mathieson.
+ * constants and functions taken from Aliroot.
+ * @author Michael Winn, Laurent Aphecetche
+ */
+
+#include "MCHSimulation/Response.h"
+
+#include "TMath.h"
+#include "TRandom.h"
+
+using namespace o2::mch;
+
+//_____________________________________________________________________
+float Response::etocharge(float edepos)
+{
+  //Todo convert in charge in number of electrons
+  //equivalent if IntPH in AliMUONResponseV0 in Aliroot
+  //to be clarified:
+  //1) why effective parameterisation with Log?
+  //2) any will to provide random numbers
+  //3) Float in aliroot, Double needed?
+  //with central seed to be reproducible?
+  //TODO: dependence on station
+  //TODO: check slope meaning in thesis
+  int nel = int(edepos * 1.e9 / 27.4);
+  float charge = 0;
+  if (nel == 0)
+    nel = 1;
+  for (int i = 1; i <= nel; i++) {
+    float arg = 0.;
+    while (!arg)
+      arg = gRandom->Rndm();
+    charge -= mChargeSlope * TMath::Log(arg);
+  }
+  //translate to fC roughly, equivalent to AliMUONConstants::DefaultADC2MV()*AliMUONConstants::DefaultA0()*AliMUONConstants::DefaultCapa() multiplication in aliroot
+  charge *= 0.61 * 1.25 * 0.2; // put this in header as constants?
+  return charge;
+}
+//_____________________________________________________________________
+double Response::chargePad(float xmin, float xmax, float ymin, float ymax, int detID, float charge)
```

Comment:
In a next PR I believe we should separate the charge integration (aka Mathieson integral) from the response (the response uses the integration). That way it will be easier to test/optimize the charge integration w/o touching at the rest.

Reply:
Ok, although I am not sure to understand 100%: You mean that you want to have two different classes? By now the two member already separate the two steps. I have nothing against such a separation for the next time.